esrally/utils/io.py

```diff
@@ -274,24 +285,55 @@ def decompress(zip_name, target_directory):
     this function.
     """
     path_without_extension, extension = splitext(zip_name)
-    filename = basename(path_without_extension)
     if extension == ".zip":
         _do_decompress(target_directory, zipfile.ZipFile(zip_name))
     elif extension == ".bz2":
-        _do_decompress_manually(target_directory, filename, bz2.open(zip_name))
+        decompressor_args = ["pbzip2", "-d", "-k", "-m10000", "-c"]
+        decompressor_lib = bz2.open(zip_name)
+        _do_decompress_manually(target_directory, zip_name, decompressor_args, decompressor_lib)
     elif extension == ".gz":
-        _do_decompress_manually(target_directory, filename, gzip.open(zip_name))
+        decompressor_args = ["pigz", "-d", "-k", "-c"]
+        decompressor_lib = gzip.open(zip_name)
+        _do_decompress_manually(target_directory, zip_name, decompressor_args, decompressor_lib)
     elif extension in [".tar", ".tar.gz", ".tgz", ".tar.bz2"]:
         _do_decompress(target_directory, tarfile.open(zip_name))
     else:
         raise RuntimeError("Unsupported file extension [%s]. Cannot decompress [%s]" % (extension, zip_name))
 
 
-def _do_decompress_manually(target_directory, filename, compressed_file):
+def _do_decompress_manually(target_directory, filename, decompressor_args, decompressor_lib):
+    decompressor_bin = decompressor_args[0]
+    base_path_without_extension = basename(splitext(filename)[0])
+
+    if is_executable(decompressor_bin):
+        if _do_decompress_manually_external(target_directory, filename, base_path_without_extension, decompressor_args):
+            return
+    else:
+        console.warn(f"{decompressor_bin} not found in PATH. Using default library, decompression will take longer.")
+
+    _do_decompress_manually_with_lib(target_directory, filename, decompressor_lib)
+
+
+def _do_decompress_manually_external(target_directory, filename, base_path_without_extension, decompressor_args):
+    logger = logging.getLogger(__name__)
+
+    with open(os.path.join(target_directory, base_path_without_extension), "wb") as new_file:
+        try:
+            subprocess.run(decompressor_args + [filename], stdout=new_file, stderr=subprocess.PIPE, check=True)
+        except subprocess.CalledProcessError as err:
+            console.warn(f"Failed to decompress [{filename}] with [{err.cmd}]. Falling back to default library.")
```

Comment:
`Falling back to standard library`?. Also, `console.warn` allows to pass a `logger` instance so maybe that would be sufficient. I also think the log statement below cannot be understood properly in isolation (i.e. without also reading the console message).

Reply:
Addressed in 746fb50c5d0c8f82a95223410bf993782dcc0044